metaflow/plugins/cards/card_decorator.py

```diff
@@ -0,0 +1,246 @@
+
+    def __init__(self, *args, **kwargs):
+        super(CardDecorator, self).__init__(*args, **kwargs)
+        self._task_datastore = None
+        self._environment = None
+        self._metadata = None
+        # todo : first allow multiple decorators with a step
+
+        # self._index is useful when many decorators are stacked on top of one another.
+        self._index = None
+
+    def add_to_package(self):
+        return list(self._load_card_package())
+
+    def _load_card_package(self):
+        try:
+            import metaflow_cards
+        except ImportError:
+            metaflow_cards_root = None
+        else:
+            metaflow_cards_root = os.path.dirname(metaflow_cards.__file__)
+
+        from . import card_modules
+
+        card_modules_root = os.path.dirname(card_modules.__file__)
+
+        if metaflow_cards_root:
+            # What if a file is too large and
+            # gets tagged along the metaflow_cards
+            # path; In such cases we can have huge tarballs
+            # that get created;
+            # Should we have package suffixes added over here?
+            for path_tuple in self._walk(metaflow_cards_root):
+                yield path_tuple
+
+        for path_tuple in self._walk(card_modules_root):
+            file_path, arcname = path_tuple
+            yield (file_path, os.path.join("metaflow", "plugins", "cards", arcname))
+
+    def _walk(self, root):
+        root = to_unicode(root)  # handle files/folder with non ascii chars
+        prefixlen = len("%s/" % os.path.dirname(root))
+        for path, dirs, files in os.walk(root):
+            for fname in files:
+                # ignoring filesnames which are hidden;
+                # TODO : Should we ignore hidden filenames
+                if fname[0] == ".":
+                    continue
+
+                p = os.path.join(path, fname)
+                yield p, p[prefixlen:]
+
+    def step_init(
+        self, flow, graph, step_name, decorators, environment, flow_datastore, logger
+    ):
+        # We do this because of Py3 support JSONDecodeError and
+        # Py2 raises ValueError
+        # https://stackoverflow.com/questions/53355389/python-2-3-compatibility-issue-with-exception
+        try:
+            import json
+
+            RaisingError = json.decoder.JSONDecodeError
+        except AttributeError:  # Python 2
+            RaisingError = ValueError
+
+        self.card_options = None
+
+        if type(self.attributes["options"]) is str:
+            # print("found attributed",self.attributes['options'],type(self.attributes['options']))
+            try:
+                self.card_options = json.loads(self.attributes["options"])
+            except RaisingError:
+                print("error loading attr")
+                # Setting Options from defaults
+                self.card_options = self.defaults["options"]
+        else:
+            self.card_options = self.attributes["options"]
```

Comment:
Can you add some comment why self.attributes["options"] is sometimes json, sometimes something else (a dict?)?